lib/aws_log_cleaner/aws_credentials.rb

```diff
@@ -9,17 +9,58 @@ class AwsCredentials
 
     attr_reader :credentials, :region
 
-    def initialize(access_key = nil, secret = nil, region = nil)
-      key_id = access_key || ENV['AWS_ACCESS_KEY_ID']
-      secret = secret || ENV['AWS_SECRET_ACCESS_KEY']
-      @region = region || ENV['AWS_DEFAULT_REGION']
+    def initialize(credential_args = nil)      
+      key_id, secret, profile = parse_credential_arguments(credential_args)
 
+      set_credentials(key_id, secret, profile)
+
+      return @credentials
+    end
+
+    private
+
+    def raise_credentials_error
+      raise "Could not load credentials."
+    end
+
+    def parse_credential_arguments(credential_args)
```

Comment:
Based on the method name, I do not expect that `@region` is set here. I think to be more consistent, we could just return `region` too, and then pass it to `set_credentials` as well.